### ISSUE TITLE
Add operation on the element of the list is not working properly

### ISSUE THREAD

**Issue by weilewei** [STEllAR-GROUP/phylanx#523]
Add operation on the element of the list is concatenating things instead of adding up element's value
I expect `[2, 1]` to be returned by the following code
```py
@Phylanx(debug=True)
def foo(x, y):
    local_x = x
    local_x[0] += y
    return local_x


a = [1, 1]
b = 1
c = foo(a, b)
print(c)
```
however, it yelds,
```
[1, 1, 1]
```
FYI, PhySL version code looks like this:
```
define(
    foo,
    x,
    y,
    block(
        define(local_x, x),
        store(local_x, local_x+y),
        local_x
    )
)
[1, 1, 1]
```
Thus, the following test failed
```py
@Phylanx
def foo(x, y):
    local_x = x
    local_x[0] += y
    return local_x


a = [1, 1]
b = 1
c = foo(a, b)
assert (c[0] == 2)
```

**Comment by weilewei** [comment]
Similarly, if I put `np.array` type element inside the list, the add operation failed. 
The following code, which is failing at this point, could be added as another testing code
```py
@Phylanx
def foo(x, y):
    local_x = x
    local_x[0] += y
    return local_x


a = [np.array([[2], [2]]), np.array([[4], [4]])]
b = np.array([[1], [1]])
c = foo(a, b)
v_0 = c[0] == np.array([[3], [3]])
assert(True == v_0.any())
```

**Comment by hkaiser** [comment]
@weilewei Both tests pass (for me) on the fixing_compiler branch, have you tried that?

**Comment by hkaiser** [comment]
@weilewei you were right today, this test is still failing. But this is unrelated to the compiler changes in #515. @rtohid this is more of an issue for the frontend.

**Comment by weilewei** [comment]
@hkaiser this issue is pretty similar to #530, at least to me. I discussed with @rtohid this afternoon, we don't have much solutions yet, but he has some ideas how to fix #530. Thanks.

**Comment by hkaiser** [comment]
@weilewei on current master I see:
```
define(
    foo,
    x,
    y,
    block(
        define(local_x, x),
        store(
            slice(local_x, 0),
            __add(
                slice(local_x, 0),
                y
            )
        ),
        local_x
    )
)
[2.0, 1]
```
which is what we should expect to see. I'm going to close this issue, please reopen, if needed.
